src/runtime/chan.go

```diff
@@ -37,11 +37,78 @@ func chanDebug(ch *channel) {
 	}
 }
 
+// blockchain is a chain of blocked operations
+type blockchain struct {
```

Comment:
While it is a funny name, I would prefer something that more closely describes what it is (and doesn't have a different but totally unrelated meaning). From other code, it sounds like this is a linked list of tasks that is blocked on a single channel. What about `channelBlockList`?